### DISCUSSION TITLE
JS client without websockets (may do only active polling by fetch())

### DISCUSSION THREAD

**Discussion by sutyrin** [colyseus/colyseus#696] [Q&A]
Hey everyone, thanks for a great project! 

Is there an API (or transport?) to implement JS client who would actively poll server for incoming updates (and post outgoing updates if need be)? I've looked up in docs and seen no such thing yet, the closest solution is to go for custom HTTP endpoints what would work with internal server methods.

This is a limitation of a platform I am developing a client for, whereas the server would be self-hosted. 

I'd be glad for any suggestions!

**Comment by endel** [comment]
Hi @sutyrin, thanks for the kind words 💙

Funnily enough, I was thinking about the possibility of this feature last week, during my experiments with WebTransport.

I think both the internal server APIs and client-side APIs are almost ready to support this feature, though they are all very tied to WebSockets right now - some structural changes are still required for both WebTransport and a potential raw HTTP/fetch transport.

For reference, this is what I have for WebTransport so far: [client-side](https://github.com/colyseus/colyseus.js/blob/webtransport/src/transport/H3Transport.ts) / [server-side](https://github.com/colyseus/colyseus/tree/webtransport/packages/transport/h3-transport/src)

I'm curious on your use-case, though - can you share a little bit about it? Cheers!

**Comment by sutyrin** [comment]
Thanks for the details!

This is a new, very beta platform, which, though, could support WS soon.

Funny that you call HTTP/fetch _raw_ compared to (web)sockets, I believe, it's the other way round ;)
